### PR TITLE
Revise Trading model

### PR DESCRIPTION
**Pull request by twzjwang**

### CODE REVIEW

File: paper.tex [hunk 1]
```diff
@@ -107,20 +108,48 @@ \section{Trading Model}
...
 
 \subsection{Participants Registration}
-At the beginning, the registrar creates a Registration Contact, which maintains all participants' information, including their DID (Decentralized Identifiers) information and public keys. Then some preselected brokers who are trustworthy are added to the decentralized data marketplace. Anyone who would like to sell or purchase data may register to become the data providers or consumers. The registrar has the authority to agree with applications.
+At the beginning, the registrar creates a Registration Contact (Figure \ref{fig:registration_contract}), which maintains all participants' information, including their DID (Decentralized Identifiers) information and public keys. Then some preselected brokers who are trustworthy are added to the decentralized data marketplace. Anyone who would like to sell or purchase data may register to become the data providers or consumers. The registrar has the authority to agree with applications.
```
Comment (jserv) [inline paper.tex]: We shall not refer to "DID information". Use "DID Document" instead.

Comment (jserv) [reply paper.tex]: I am not sure if "preselected brokers" is a meaningful term. How about something like Committee of Conformity Assessment?

Comment (jkrvivian) [inline paper.tex]: Typo, `Registration Contract`

File: paper.tex [hunk 2]
```diff
@@ -107,20 +108,48 @@ \section{Trading Model}
...
+Brokers certify session keys as well. However, it is a risk revealing session keys to brokers since contents may be copied by brokers, which causes data providers' loss. Therefore, Blind signature is used to prevent the situation. Figure \ref{fig:key_certification} shows the certification process. When a data provider asks a broker to ceritfy new session key, the data provider uses the broker's public key, which is available in the Registration Contact after the broker's registration, as a blinding factor, and the session key is blinded. Then the data provider sends the blinded session key to the broker. The broker signs the message and returns the signature to data provider. The data provider removes the blinding factor and obtains the broker's signature of the session key. Only one session key can be signed in each product, so data providers can't fake a session key to deceive consumers.
```
Comment (jkrvivian) [inline paper.tex]: Typo, `, which is available in the Registration Contract...`, " Contract"

File: paper.tex [hunk 3]
```diff
@@ -107,20 +108,48 @@ \section{Trading Model}
...
...cribe the sensor data pays a subscription fee to the Product Contract, it is added to the consumer list automatically by the smart contract. To decrypt the streaming data on the MAM channel, session key should be exchanged between data provider and consumers as shown in Figure \ref{fig:key_exchange}. The data provider requests the subscription list maintained by the Product Contract and obtains accounts of each consumer. After that, the data provider can also get public keys of each consumer from the Registration Contact with their accounts. For each consumer, the data provider encrypts the session key and broker's signature with the consumer's public key and sends the ciphertext to the Product Contract. Consumers listen to the smart contract event which is triggered when the ciphertext is updated, and decrypt the ciphertext to get the session key and signature. Consumers can obtain the broker's public key on the Registration Contact as well, so they can verify that the signature is valid and the session key is the only one that is certified by the broker. Afterward, encrypted sensing data is uploaded to the MAM channel, and consumers can download and decrypt them with the session¸
```
Comment (jkrvivian) [inline paper.tex]: The last sentence is not finished, `and decrypt them with the session key.`, `key` is missing here.

**Pull request merged**